internal/jimm/config/config.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2025 Canonical.
+
+package config
+
+import (
+	jujucontroller "github.com/juju/juju/controller"
+	jujuparams "github.com/juju/juju/rpc/params"
+)
+
+// configManager exposes the configuration of the JIMM controller when
+// the ControllerConfig facade is called on the controller root.
+// At the moment it only exposes a map, but in the future we could fetch
+// the configuration from the database or other sources.
+type configManager struct {
+	config ControllerConfig
+}
+
+// ControllerConfig holds the configuration to be returned when the ControllerConfig
+// facade is called on the controller root.
+type ControllerConfig struct {
+	// ControllerUUID is the UUID of the JIMM controller.
+	ControllerUUID string
+
+	// PublicDNSName is returned to Juju clients on login.
+	// It is the hostname that will be used during TLS verification.
+	PublicDNSName string
+
+	// APIPort is the port for API connections.
+	APIPort int
+
+	// SSHPort is the port for SSH connections.
+	SSHPort int
+
+	// SSHMaxConcurrentConnections is the maximum number of concurrent SSH connections.
+	SSHMaxConcurrentConnections int
+
+	// SSHPublicHostKey is the host key for SSH connections.
+	SSHPublicHostKey string
+}
+
+// NewConfigManager creates a new config manager with the given configuration.
+func NewConfigManager(config ControllerConfig) (*configManager, error) {
+	return &configManager{
+		config: config,
+	}, nil
+}
+
+// GetConfig returns the configuration for the controller converted in the jujuparams format.
+func (c *configManager) GetConfig() (jujuparams.ControllerConfig, error) {
```

Comment:
This manager shouldn't return a jujuparams.ControllerConfig (As it's a DTO), this is an API layer param, to keep the concerns separated just return configManager.config & then in the facade, map it over.

Reply:
+1 to this ^^